### ISSUE TITLE
-Wsometimes-uninitialized in drivers/gpu/drm/amd/amdgpu/amdgpu_ras.c

### ISSUE THREAD

**Issue by nathanchance** [ClangBuiltLinux/linux#1595]
Initially reported by the kernel test robot: https://lore.kernel.org/r/202202171534.A4Lb31Xv-lkp@intel.com/

```
drivers/gpu/drm/amd/amdgpu/amdgpu_ras.c:2416:7: error: variable 'ras_obj' is used uninitialized whenever 'if' condition is true [-Werror,-Wsometimes-uninitialized]
                if (adev->in_suspend || amdgpu_in_reset(adev)) {
                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
drivers/gpu/drm/amd/amdgpu/amdgpu_ras.c:2453:6: note: uninitialized use occurs here
        if (ras_obj->ras_cb)
            ^~~~~~~
drivers/gpu/drm/amd/amdgpu/amdgpu_ras.c:2416:3: note: remove the 'if' if its condition is always false
                if (adev->in_suspend || amdgpu_in_reset(adev)) {
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
drivers/gpu/drm/amd/amdgpu/amdgpu_ras.c:2416:7: error: variable 'ras_obj' is used uninitialized whenever '||' condition is true [-Werror,-Wsometimes-uninitialized]
                if (adev->in_suspend || amdgpu_in_reset(adev)) {
                    ^~~~~~~~~~~~~~~~
drivers/gpu/drm/amd/amdgpu/amdgpu_ras.c:2453:6: note: uninitialized use occurs here
        if (ras_obj->ras_cb)
            ^~~~~~~
drivers/gpu/drm/amd/amdgpu/amdgpu_ras.c:2416:7: note: remove the '||' if its condition is always false
                if (adev->in_suspend || amdgpu_in_reset(adev)) {
                    ^~~~~~~~~~~~~~~~~~~
drivers/gpu/drm/amd/amdgpu/amdgpu_ras.c:2403:41: note: initialize the variable 'ras_obj' to silence this warning
        struct amdgpu_ras_block_object *ras_obj;
                                               ^
                                                = NULL
2 errors generated.
```

Tom Rix submitted: https://lore.kernel.org/r/20220217153842.3252424-1-trix@redhat.com/

**Comment by nathanchance** [comment]
Patch accepted: https://gitlab.freedesktop.org/agd5f/linux/-/commit/779596ce6a79e187995f04f143fc5ea44a565ea9

**Comment by nathanchance** [comment]
Fixed in `next-20220222`: https://git.kernel.org/next/linux-next/c/779596ce6a79e187995f04f143fc5ea44a565ea9
